odbc/src/api/functions.rs

```diff
@@ -1948,16 +1991,39 @@ pub unsafe extern "C" fn SQLRowCount(
 /// [`SQLSetConnectAttr`]: https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/SQLSetConnectAttr-function
 ///
 /// # Safety
-/// Because this is a C-infereface, this is necessarily unsafe
+/// Because this is a C-interface, this is necessarily unsafe
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLSetConnectAttr(
```

Comment:
I think we might as well support setting the current catalog since it could prove useful for testing (and eventually we'll want it anyway).

Reply:
Also move this code to `SQLSetConnectAttrW` for reasons as above. It will matter once you support setting the current catalog since that is going to be passed as a `*WChar`.

My latest patch also adds support for ASCII strings, but we don't need to support _accepting_ ASCII strings yet, imo.